agent/app/agent.go

```diff
@@ -751,6 +747,37 @@ func (agent *ecsAgent) constructVPCSubnetAttributes() []*ecs.Attribute {
 	}
 }
 
+// Loads Managed Daemon images for all Managed Daemons registered on the Agent.
+// The images are loaded in the background. Successfully loaded images are added to
+// imageManager's cleanup exclusion list.
+func (agent *ecsAgent) loadManagedDaemonImagesAsync(imageManager engine.ImageManager) {
+	daemonManagers := agent.getDaemonManagers()
+	logger.Debug(fmt.Sprintf("Will load images for %d Managed Daemons", len(daemonManagers)))
+	for _, daemonManager := range daemonManagers {
+		go agent.loadManagedDaemonImage(daemonManager, imageManager)
+	}
+}
+
+// Loads Managed Daemon image and adds it to image cleanup exclusion list upon success.
+func (agent *ecsAgent) loadManagedDaemonImage(dm dm.DaemonManager, imageManager engine.ImageManager) {
+	logger.Info("Starting to load Managed Daemon image", logger.Fields{
+		field.ImageRef: dm.GetManagedDaemon().GetImageRef(),
```

Comment:
nit: we're referencing this imageref a few times -- would it be useful to grab it once as we enter the function and use it?
```
daemonImageRef := dm.GetManagedDaemon().GetImageRef()
```
then use that everywhere below...

Reply:
Sure. I will update.

Reply:
Updated in https://github.com/aws/amazon-ecs-agent/pull/3984/commits/a56ea6195b4378a019d0016a40996720c4ece7a1